x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -61,6 +61,10 @@ export const getRuleExecutor = (): LifecycleRuleExecutor<
     const summaryClient = new DefaultSLIClient(esClient.asCurrentUser);
     const slo = await sloRepository.findById(params.sloId);
 
+    if (!slo.enabled) {
```

Comment:
It basically returns a no state response. Meaning there are no active alerts, nor recovered alerts for this execution.

Reply:
I see, so there is no way to skip executing the executor when a rule is disabled, right?

Reply:
There is an internal bulk enable/disable api for rules. But I'm not sure if there is a service available from the backend directly. So it would require a first call to our API, and then a call to the internal rule API to disable the related rules. 

That's possible, but then nothing prevent the customer to enable the rule directly from the rules page, so we still need a way to shortcut the rule executor.

Reply:
I see your point. Indeed, it is good to have this check here.

I still think it is beneficial to have a link between disabling an SLO and disabling the related rules; then, there is no need to execute rules that their SLO is disabled on the specified interval. Also, it will be more clear to the user that even though this rule is enabled, the execution will not do anything.

Maybe when disabling an SLO, we can check if there is any related rule and ask users whether they want to disable those as well or not. Up to you and the product team to decide :) 
 
cc @grabowskit @katrin-freihofner

Reply:
> I still think it is beneficial to have a link between disabling an SLO and disabling the related rules; then, there is no need to execute rules that their SLO is disabled on the specified interval. Also, it will be more clear to the user that even though this rule is enabled, the execution will not do anything.
> Maybe when disabling an SLO, we can check if there is any related rule and ask users whether they want to disable those as well or not. Up to you and the product team to decide :)

I agree. We don't have a user's facing way to disable/enable an SLO yet. It's only accessible from the API for now. But when we do expose this, it's a good idea to tell the user to go disable the rules.

Reply:
> it's a good idea to tell the user to go disable the rules.

Or we do it automatically for them.